portiaui/app/services/annotation-structure.js

```diff
@@ -146,14 +147,39 @@ const ElementStructure = Ember.Object.extend({
                         selectorMatcher.unRegister(selector, setElements);
                     }
                     selector = annotation.get('selectorGenerator.selector');
+                    const repeatedSelector = annotation.get('containerGenerator.selector');
                     if (!annotation.get('isDeleted')) {
                         annotation.setProperties({
                             selector,
                             xpath: annotation.get('selectorGenerator.xpath')
                         });
                     }
+                    annotation.set('repeated', false);
                     if (selector) {
                         selectorMatcher.register(selector, setElements);
+                        const elements = selectorMatcher.query(selector);
+                        if (repeatedSelector && elements && elements.length > 1) {
+                            const repeatedContainers = annotation.get('containerGenerator.parent.repeatedContainers');
+                            if (repeatedContainers.length > 1) {
+                                for (let container of repeatedContainers) {
+                                    let i = 0;
+                                    for (let child of elements) {
+                                        if (container.contains(child)) {
+                                            i += 1;
+                                        }
+                                        if (i > 1) {
+                                            break;
+                                        }
+                                    }
+                                    if (i > 1) {
+                                        annotation.set('repeated', true);
+                                        break;
+                                    }
+                                }
```

Comment:
Extract the inner function out?

`repeatedContainers.map(containedChildren).any((repeated) => { repeated.length > 1 });`